integration/spark/shared/src/main/java/io/openlineage/spark/agent/lifecycle/plan/LogicalRelationDatasetBuilder.java

```diff
@@ -100,6 +103,8 @@ public List<D> apply(LogicalRelation logRel) {
       return handleHadoopFsRelation(logRel);
     } else if (logRel.relation() instanceof JDBCRelation) {
       return new JdbcRelationHandler<>(datasetFactory).handleRelation(logRel);
+    } else if (logRel.relation() instanceof LineageRelation) {
+      return new BuiltInLineageV1RelationHandler<>(context, datasetFactory).handleRelation(logRel);
```

Comment:
This relation handler should always be the first checked - both here and in `DatasetBuilderFactories`. When extension provides lineage, it should take precedence over our internal implementations.

Reply:
Fixed here, but not sure if this makes sense in factories `Spark34DatasetBuilderFactory`. First, I don't think order of builders is guaranteed. Secondly, we always run all the builders.

Reply:
>First, I don't think order of builders is guaranteed.

In this case I think we should guarantee that.